pontoon/base/tests/formats/test_po.py

```diff
@@ -0,0 +1,385 @@
+            resource.translations[7],
+            comments=[],
+            source=[],
+            key='Plural %(count)s string with missing translations',
+            source_string='Plural %(count)s string with missing translations',
+            source_string_plural='Plural %(count)s strings with missing translations',
+            strings={
+                1: 'Translated Plural %(count)s strings with missing translations'
+            },
+            fuzzy=False,
+            order=7,
+        )
+
+    def generate_pofile(
+        self, body,
+        revision_date='2015-08-04 12:30+0000',
+        last_translator='example <example@example.com>',
+        language='test_locale',
+        plural_forms='nplurals=2; plural=(n != 1);',
+        generator='Pontoon'
+    ):
+        header = HEADER_TEMPLATE.format(
+            revision_date=revision_date,
+            last_translator=last_translator,
+            language=language,
+            plural_forms=plural_forms,
+            generator=generator
+        )
+        return header + body
+
+    def assert_pofile_equal(self, pofile_path, expected_content):
+        with open(pofile_path) as f:
+            self.assertMultiLineEqual(f.read(), expected_content)
+
+    def test_save_basic(self):
+        """
+        Test saving changes to an entity with a single translation.
+        """
+        test_input = self.generate_pofile(dedent("""
+            #. Comment
+            #: file.py:1
+            msgid "Source String"
+            msgstr "Translated String"
+        """))
+        path, resource = self.parse_string(test_input)
+
+        translation = resource.translations[0]
+        translation.strings[None] = 'New Translated String'
+        translation.fuzzy = True
+        resource.save(self.locale)
+
+        self.assert_pofile_equal(path, self.generate_pofile(dedent("""
+            #. Comment
+            #: file.py:1
+            #, fuzzy
+            msgid "Source String"
+            msgstr "New Translated String"
+        """)))
+
+    def test_save_plural(self):
+        test_input = self.generate_pofile(dedent("""
+            msgid "Plural %(count)s string"
+            msgid_plural "Plural %(count)s strings"
+            msgstr[0] "Translated Plural %(count)s string"
+            msgstr[1] "Translated Plural %(count)s strings"
+        """))
+        path, resource = self.parse_string(test_input)
+
+        translation = resource.translations[0]
+        translation.strings[0] = 'New Plural'
+        translation.strings[1] = 'New Plurals'
+        resource.save(self.locale)
+
+        self.assert_pofile_equal(path, self.generate_pofile(dedent("""
+            msgid "Plural %(count)s string"
+            msgid_plural "Plural %(count)s strings"
+            msgstr[0] "New Plural"
+            msgstr[1] "New Plurals"
+        """)))
+
+    def test_save_plural_remove(self):
+        """Any missing plurals should be removed from the pofile."""
```

Comment:
Not like this. You need to keep the length of msgstr to be nplurals, otherwise PO file will not be valid. When deleting strings, you basically need to set them to empty string: "".